dynamiq/evaluations/metrics/faithfulness.py

```diff
@@ -0,0 +1,257 @@
+                                "degree in Computer Science. He is enrolled in several "
+                                "courses this semester, including Data Structures, Algorithms, "
+                                "and Database Management. John is a diligent student and "
+                                "spends a significant amount of time studying and completing "
+                                "assignments. He often stays late in the library to work on "
+                                "his projects."
+                            )
+                        ],
+                        "statements": [
+                            [
+                                "John is majoring in Biology.",
+                                "John is taking a course on Artificial Intelligence.",
+                                "John is a dedicated student.",
+                                "John has a part-time job.",
+                            ]
+                        ],
+                    },
+                    "outputs": {
+                        "results": [
+                            {
+                                "statement": "John is majoring in Biology.",
+                                "verdict": 0,
+                                "reason": (
+                                    "The context states that John is pursuing a degree "
+                                    "in Computer Science, not Biology."
+                                ),
+                            },
+                            {
+                                "statement": "John is taking a course on Artificial Intelligence.",
+                                "verdict": 0,
+                                "reason": (
+                                    "The context lists his courses, and Artificial " "Intelligence is not mentioned."
+                                ),
+                            },
+                            {
+                                "statement": "John is a dedicated student.",
+                                "verdict": 1,
+                                "reason": (
+                                    "The context mentions he spends significant time studying "
+                                    "and stays late to work on projects."
+                                ),
+                            },
+                            {
+                                "statement": "John has a part-time job.",
+                                "verdict": 0,
+                                "reason": (
+                                    "There is no information in the context about John " "having a part-time job."
+                                ),
+                            },
+                        ]
+                    },
+                },
+            ],
+            llm=self.llm,
+        )
+
+    def evaluate(
+        self,
+        questions: list[str],
```

Comment:
Please check if it is possible to unify evaluate/run parameters and their order. I some I see `contexts`, in other `contexts_list`.

Reply:
I tried to unify the names and order of the parameters.

However, the exact parameters will differ for each metric. Some metrics require ground truth, while others don't need it but instead require context, for instance.